File: app/src/main/java/org/coscup/ccip/adapter/SpeakerImageAdapter.java
package org.coscup.ccip.adapter;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

import org.coscup.ccip.fragment.SpeakerFragment;
import org.coscup.ccip.model.Speaker;

import java.util.ArrayList;
import java.util.List;

public class SpeakerImageAdapter extends FragmentStatePagerAdapter {
    private List<Speaker> speakers = new ArrayList<>();
    private List<Fragment> mFragmentList = new ArrayList<>();

    public SpeakerImageAdapter(FragmentManager fm, List<Speaker> speakers) {
        super(fm);
        this.speakers = speakers;
        for (Speaker speaker : this.speakers) {
            mFragmentList.add(SpeakerFragment.newInstance(speaker));
        }
    }

    @Override
    public int getCount() {
        return this.mFragmentList.size();
    }

    @Override
    public Fragment getItem(int position) {
        return mFragmentList.get(position);
    }
}

File: app/src/main/java/app/opass/ccip/ui/sessiondetail/SpeakerImageAdapter.kt
package app.opass.ccip.ui.sessiondetail

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import app.opass.ccip.model.Speaker

class SpeakerImageAdapter(fm: FragmentManager, speakers: List<Speaker>) : FragmentStatePagerAdapter(fm) {
    private val mFragmentList = ArrayList<Fragment>()

    init {
        for (speaker in speakers) {
            mFragmentList.add(SpeakerFragment.newInstance(speaker))
        }
    }

    override fun getCount() = this.mFragmentList.size

    override fun getItem(position: Int) = mFragmentList[position]
}
